.goreleaser.yml

```diff
@@ -0,0 +1,87 @@
+before:
+  hooks:
+    - go mod download
+    # you may remove this if you don't need go generate
+    - go generate ./...
+project_name: auth-gateway
+builds:
+  - id: auth-gateway-darwin
+    ldflags:
+      -s -w -X github.com/cortexproject/auth-gateway/version.Version={{.Version}}
+    binary: auth-gateway
+    env:
+      - CGO_ENABLED=0
+    goos:
+      - darwin
+    goarch:
+      - amd64
+      - arm64
+  - id: auth-gateway-linux
+    ldflags:
+      -s -w -X github.com/cortexproject/auth-gateway/version.Version={{.Version}}
+    binary: auth-gateway
+    env:
+      - CGO_ENABLED=0
+    goos:
+      - linux
+    goarch:
+      - amd64
+      - arm64
+  - id: auth-gateway-windows
+    ldflags:
+      -s -w -X github.com/cortexproject/auth-gateway/version.Version={{.Version}}
+    binary: auth-gateway
+    env:
+      - CGO_ENABLED=0
+    goos:
+      - windows
+    goarch:
+      - amd64
+      - arm64
+archives:
+  - format_overrides:
+      - goos: windows
+        format: zip
+    files:
+      - none*
+    format: binary
+checksum:
+  name_template: 'checksums.txt'
+snapshot:
+  name_template: "{{ .Tag }}-next"
+changelog:
+  sort: asc
+  filters:
+    exclude:
+      - '^docs:'
+      - '^test:'
+dockers:
+  - goarch: amd64
+    image_templates:
+      - "quay.io/cortexproject/auth-gateway:{{ .Tag }}-amd64"
+    build_flag_templates:
+    - "--label=org.opencontainers.image.created={{.Date}}"
+    - "--label=org.opencontainers.image.name={{.ProjectName}}"
+    - "--label=org.opencontainers.image.revision={{.FullCommit}}"
+    - "--label=org.opencontainers.image.version={{.Version}}"
+    - "--label=repository=https://github.com/cortexproject/auth-gateway"
+    - "--label=homepage=https://cortexmetrics.io"
+    - "--platform=linux/amd64"
+    use: buildx
+  - goarch: arm64
+    image_templates:
+      - "quay.io/cortexproject/auth-gateway:{{ .Tag }}-arm64"
+    build_flag_templates:
+    - "--label=org.opencontainers.image.created={{.Date}}"
+    - "--label=org.opencontainers.image.name={{.ProjectName}}"
+    - "--label=org.opencontainers.image.revision={{.FullCommit}}"
+    - "--label=org.opencontainers.image.version={{.Version}}"
+    - "--label=repository=https://github.com/cortexproject/auth-gateway"
+    - "--label=homepage=https://cortexmetrics.io"
+    - "--platform=linux/arm64"
+    use: buildx
+docker_manifests:
+    - name_template: "quay.io/cortexproject/auth-gateway:{{.Tag}}"
+      image_templates:
+      - "quay.io/cortexproject/auth-gateway:{{.Tag}}-amd64"
+      - "quay.io/cortexproject/auth-gateway:{{.Tag}}-arm64"
```

Comment:
Combines both images into one, so that one image can be used for amd64 and arm64

Reply:
Okay, I run the command you said but this line could not get executed:

`RUN        apk add --update --no-cache ca-certificates`

Here is the trace:

```
  • starting release...
  • loading config file                              file=.goreleaser.yml
  • loading environment variables
  • getting and validating git state
    • ignoring errors because this is a snapshot     error=git doesn't contain any tags. Either add a tag or use --snapshot
    • building...                                    commit=1793cedb2b39a4bb8370a5faa8aa451d0b14d98d latest tag=v0.0.0
    • pipe skipped                                   reason=disabled during snapshot mode
  • parsing tag
  • setting defaults
  • running before hooks
    • running                                        hook=go mod download
    • running                                        hook=go generate ./...
  • snapshotting
    • building snapshot...                           version=v0.0.0-next
  • checking distribution directory
    • cleaning dist
  • loading go mod information
  • build prerequisites
  • writing effective config file
    • writing                                        config=dist/config.yaml
  • building binaries
    • building                                       binary=dist/auth-gateway-linux_linux_arm64/auth-gateway
    • building                                       binary=dist/auth-gateway-darwin_darwin_arm64/auth-gateway
    • building                                       binary=dist/auth-gateway-linux_linux_amd64_v1/auth-gateway
    • building                                       binary=dist/auth-gateway-darwin_darwin_amd64_v1/auth-gateway
    • building                                       binary=dist/auth-gateway-windows_windows_amd64_v1/auth-gateway.exe
    • building                                       binary=dist/auth-gateway-windows_windows_arm64/auth-gateway.exe
    • took: 2s
  • archives
    • creating                                       archive=dist/auth-gateway_v0.0.0-next_windows_amd64.zip
    • skip archiving                                 binary=auth-gateway name=auth-gateway_v0.0.0-next_linux_amd64
    • skip archiving                                 binary=auth-gateway name=auth-gateway_v0.0.0-next_darwin_amd64
    • skip archiving                                 binary=auth-gateway name=auth-gateway_v0.0.0-next_linux_arm64
    • creating                                       archive=dist/auth-gateway_v0.0.0-next_windows_arm64.zip
    • skip archiving                                 binary=auth-gateway name=auth-gateway_v0.0.0-next_darwin_arm64
    • no files matched                               glob=none*
    • no files matched                               glob=none*
    • took: 1s
  • calculating checksums
  • docker images
    • building docker image                          image=quay.io/cortexproject/auth-gateway:v0.0.0-arm64
    • building docker image                          image=quay.io/cortexproject/auth-gateway:v0.0.0-amd64
    • took: 1s
  ⨯ release failed after 4s                          error=docker build failed: failed to build quay.io/cortexproject/auth-gateway:v0.0.0-arm64: exit status 1: #0 building with "default" instance using docker driver

#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 216B done
#1 DONE 0.0s

#2 [internal] load .dockerignore
#2 transferring context: 2B done
#2 DONE 0.1s

#3 [internal] load metadata for docker.io/library/alpine:3.18
#3 DONE 0.6s

#4 [1/3] FROM docker.io/library/alpine:3.18@sha256:82d1e9d7ed48a7523bdebc18cf6290bdb97b82302a8a9c27d4fe885949ea94d1
#4 CACHED

#5 [internal] load build context
#5 transferring context: 9.77MB 0.1s done
#5 DONE 0.2s

#6 [2/3] RUN        apk add --update --no-cache ca-certificates
#6 0.264 exec /bin/sh: exec format error
#6 ERROR: process "/bin/sh -c apk add --update --no-cache ca-certificates" did not complete successfully: exit code: 1
------
 > [2/3] RUN        apk add --update --no-cache ca-certificates:
0.264 exec /bin/sh: exec format error
------
Dockerfile:2
--------------------
   1 |     FROM       alpine:3.18
   2 | >>> RUN        apk add --update --no-cache ca-certificates
   3 |     COPY       auth-gateway /usr/bin/auth-gateway
   4 |     EXPOSE     80
--------------------
ERROR: failed to solve: process "/bin/sh -c apk add --update --no-cache ca-certificates" did not complete successfully: exit code: 1

Learn more at https://goreleaser.com/errors/docker-build

```